Tutorial-Learn-how-to-make-a-backend-change-that-has-an-impact-on-server-data.md

```diff
@@ -0,0 +1,677 @@
+2. **Network Inspection**: Observe the network call triggered when changes are made on the Preferences page to trace the update process through the codebase.
+
+In this tutorial, we’ll use the second approach to showcase how to trace functionality effectively.
+
+> [!IMPORTANT]
+> Practice 1: Familiarize yourself with the codebase architecture at Oppia. Understanding the structure of the codebase will help you navigate through various layers of code at Oppia more efficiently. Follow this guide: [Overview of the Oppia Codebase](https://github.com/oppia/oppia/wiki/Overview-of-the-Oppia-codebase).
+
+![Screenshot of Preference Page](images/TutorialMigrationJob/preferencePage.png)
+
+Above is an image of the Preferences page, which includes various fields. In our case, we need to update the `Bio` field, then click the **Save Changes** button and observe which endpoint is triggered in the network tab of your browser's developer tools.
+
+1. Enter any random text in the `Bio` field on the Preferences page. Notice that the **Save Changes** button becomes clickable (it’s disabled by default until changes are made).  
+2. Click the **Save Changes** button and open the **Network Tab** in your browser’s developer tools.  
+3. Clear any existing network calls to focus on new ones.
+
+> [!IMPORTANT]
+> Practice 2: From the network tab in your browser's developer tools, can you figure out which network call is made to update the `user_bio` field at Oppia?
+> 
+> **Hint**: When you first open the network tab, you might see a large number of network calls. To narrow them down, refresh the network tab just before clicking the Save Changes button. This will help you focus on the new calls triggered by the action. For a deeper understanding of how to use Chrome DevTools effectively, refer to this guide: [Chrome DevTools Network Panel Documentation](https://developer.chrome.com/docs/devtools/network).
+
+Upon clicking the **Save Changes** button, you’ll notice a call to the following endpoint:  
+`http://localhost:8181/preferenceshandler/data`
+
+This tells us that the endpoint (`/preferenceshandler/data`) handles updates to the Preferences page.
+
+![Screenshot of Preference Page with Network Tab opened](images/TutorialMigrationJob/preferencePageWithNetworkTab.png)
+
+Now it’s time to trace this endpoint in the codebase.
+
+The URL triggered when clicking the **Save Changes** button is `/preferenceshandler/data`. Perform a quick search in the codebase for this endpoint. Note that exact URL matches may not yield results, as URLs are often aliased as constants. To locate the relevant code, try variations such as `/preferenceshandler/data`, `/preferenceshandler`, or `preferenceshandler`.
+
+> [!IMPORTANT]
```

Comment:
I think you should explain something before this practice about main.py containing all the routings, so it's less about "generally search the codebase" and more about "easily find the relevant routing in main.py". It's great that you have the note about the possible aliasing-by-constants.